File: src/src_core/fmg/core/types/viewmodel/event/ChampionModelEvent.cs
namespace fmg.core.types.viewmodel.Event {

    public delegate void ChampionModelChangedHandler(object sender, ChampionModelEventArgs e);

    public class ChampionModelEventArgs : System.EventArgs {

        public const int POS_ALL = -1;
        public const int INSERT = 1, UPDATE = 2, DELETE = 3;

        public EMosaic     Mosaic  { get; }
        public ESkillLevel Skill   { get; }
        public int         Pos     { get; }
        public int         Type    { get; }

        public ChampionModelEventArgs(EMosaic mosaic, ESkillLevel skill, int pos, int type) {
            this.Mosaic = mosaic;
            this.Skill = skill;
            this.Pos = pos;
            this.Type = type;
        }

    }

}

File: src/src_core/fmg/core/types/viewmodel/event/ChampionModelEvent.java
package fmg.core.types.viewmodel.event;

import java.util.EventObject;

import fmg.core.types.EMosaic;
import fmg.core.types.ESkillLevel;
import fmg.core.types.viewmodel.serializable.ChampionsModel;

public class ChampionModelEvent extends EventObject {
   private static final long serialVersionUID = 1L;

   public static final int POS_ALL = -1;
   public static final int INSERT = 1, UPDATE = 2, DELETE = 3;

   private final EMosaic mosaic;
   private final ESkillLevel skill;
   private final int pos;
   private final int type;

   public ChampionModelEvent(ChampionsModel source, EMosaic mosaic, ESkillLevel skill, int pos, int type) {
      super(source);
      this.mosaic = mosaic;
      this.skill = skill;
      this.pos = pos;
      this.type = type;
   }

   public ESkillLevel getSkill() {
      return skill;
   }
   public EMosaic getMosaic() {
      return mosaic;
   }
   public int getPos() {
      return pos;
   }
   public int getType() {
      return type;
   }

   @Override
   public ChampionsModel getSource() {
      return (ChampionsModel) super.getSource();
   }
}
